### PR TITLE
Added boost_system library to link client.

### PR DESCRIPTION
**Pull request by mverboom**
Had problems compiling the client code. As far as I can see it was only missing the boost_system library in the linking phase. Works for me after this change.

Regards,

Mark

### CODE REVIEW

**Comment by badaix** [comment]
Hi Mark, 
thanks for the PR. To be honest, I was confused why I don't have to link against boost_system (as far as I could remember, this is neccessary for asio). 
My dev machine is running Linux Mint 18.3 with gcc 5.4.0 and boost 1.70 (from ppa:mhier/libboost-latest), I'm doing ARM builds on an RPi3 running Raspbian 9.11 with gcc 6.3.0. On both I didn't need to link against boost_system. The RPi3 doesn't even have boost installed, but only downloaded and let the include path point to it: `make ADD_CFLAGS=-I/home/pi/Develop/boost_1_71_0/`
Finally, it's running on Travis with 14 different compiler versions (gcc 4.9 to 9.0, clang 3.9 to 7 and 3 different xcodes (should be some clang)), all of these builds don't link against boost_system.
Long story short: I guess it depends on the boost version. Can you please try to build w/o linking against boost system, but with `-DBOOST_ERROR_CODE_HEADER_ONLY` added to `CXXFLAGS`?

**Comment by mverboom** [comment]
Hi,

Tried this with the -DBOOST_ERROR_CODE_HEADER_ONLY and it now works!

Just for you informatin, I am building on debian buster. Build machine is very bare install. When building software is installs only the required packages for build. Don't know if that has anything to do with this needing the extra definition to build the client, but just wanted to let you know.

Still have to take a look at why it fails building the server part on buster, but won't put that here as this is a client PR.

Thanks for the quick reply and the great software, use it daily :)

Mark

**Comment by badaix** [comment]
This are good news, as the requirement to have a recent boost installed is a much harder than the requirement to just have the headers available. The last years I tried my best not to depend on boost, but I lost the fight ;). Can you change the PR accordingly and/or try out if CMake is having the same issue?
Hope you can  sort out your server compilation issues :crossed_fingers:

**Comment by badaix** [comment]
seems that you will need at least boost 1.70, see #485
The compiler definition is added to the makefiles now
